### PR TITLE
Remove DX11 backend

### PR DESCRIPTION
**Pull request by valaphee**
**Connections**
https://github.com/gfx-rs/wgpu/issues/4797

**Description**
This backend had no functionality, and is mainly just boilerplate.

**Testing**

**Checklist**

- [x] Run `cargo fmt`.
- [x] Run `cargo clippy`. If applicable, add:
  - [x] `--target wasm32-unknown-unknown`
  - [x] `--target wasm32-unknown-emscripten`
- [x] Run `cargo xtask test` to run tests.
- [x] Add change to `CHANGELOG.md`. See simple instructions inside file.

Also updated etc/big-picture.png, and all references should be removed except in the changelog, logically.

### CODE REVIEW

**Comment by valaphee** [comment]
Should I also correct the ids here? https://github.com/gfx-rs/wgpu/blob/a418b60154a9d97677ec53c3e0c497afc2e6b900/wgpu-core/src/id.rs#L133

**Comment by valaphee** [comment]
Should I also update the backend ids? https://github.com/gfx-rs/wgpu/blob/a418b60154a9d97677ec53c3e0c497afc2e6b900/wgpu-types/src/lib.rs#L106 Gl would become 4 and WebGPU = 5

**Review by crowlKats — COMMENTED** [review]
no comment for deno changes.

File: wgpu-core/src/id.rs [hunk 1]
```diff
@@ -131,7 +131,6 @@ where
             1 => Backend::Vulkan,
             2 => Backend::Metal,
             3 => Backend::Dx12,
-            4 => Backend::Dx11,
```
Comment (cwfitzgerald) [inline wgpu-core/src/id.rs:134]: Yeah, let's renumber the backends.

**Review by cwfitzgerald — APPROVED** [review]

**Pull request merged**